SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp

```diff
@@ -95,12 +97,21 @@ struct ConfigureConnection
 		Params.network.kcp.upstream_kcp.flush_interval_millis = Config.UdpUpstreamIntervalMS;
 		Params.network.kcp.downstream_kcp.flush_interval_millis = Config.UdpDownstreamIntervalMS;
 
+		const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>();
+		float TimeoutSeconds;
 #if WITH_EDITOR
-		Params.network.tcp.downstream_heartbeat = &HeartbeatParams;
-		Params.network.tcp.upstream_heartbeat = &HeartbeatParams;
-		Params.network.kcp.downstream_heartbeat = &HeartbeatParams;
-		Params.network.kcp.upstream_heartbeat = &HeartbeatParams;
+		TimeoutSeconds = Settings->HeartbeatTimeoutWithEditorSeconds;
+		
+#else
+		TimeoutSeconds = Settings->HeartbeatTimeoutSeconds;
 #endif
+		Worker_HeartbeatParameters WorkerHeartbeatParams = { static_cast<uint64_t>(Settings->HeartbeatIntervalSeconds
```

Comment:
This data is on the stack, which will get popped once the function returned, and then the `Params.network.tcp.downstream_heartbeat` will point to bad data. Re-use the existing `Worker_HeartbeatParameters HeartbeatParams` member this object already has.